applications/shell/src/lib.rs

```diff
@@ -1451,9 +1451,8 @@ impl Shell {
                 if let Some(job) = self.jobs.get_mut(&job_num) {
                     self.fg_job_num = Some(job_num);
                     for task_ref in &job.tasks {
-                        if !task_ref.has_exited() {
-                            task_ref.unblock();
-                        }
+                        let _ = task_ref.unblock();
+                        // TODO: Do we want to set the job status of an exited task to running?
```

Comment:
no, if the unblock operation fails because the task is exited here, there we should mark it as such and remove the job.